app/src/main/java/org/eyeseetea/malariacare/SettingsActivity.java

```diff
@@ -905,9 +910,29 @@ boolean isLoginDone(){
      * Launches Login activity with the right extra params
      */
     void launchLoginPreChange(){
-        Intent intent = prepareIntent();
-        activity.finish();
-        activity.startActivity(intent);
+
+        new AlertDialog.Builder(activity)
+                .setTitle(activity.getString(R.string.settings_menu_logout))
+                .setMessage(activity.getString(R.string.dialog_content_dhis_preference_login))
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface arg0, int arg1) {
+                        //finish activity and go to login
+                       // DhisService.logOutUser(activity);
+                        User.clearLoggedUser();
+
+                        Intent intent = prepareIntent();
+                       intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
```

Comment:
I thought this was automatically done by Android but in the case of a Service...are these flags really needed?